geopandas/tests/test_geom_methods.py

```diff
@@ -564,6 +569,56 @@ def test_distance_crs_warning(self):
         with pytest.warns(UserWarning, match="Geometry is in a geographic CRS"):
             self.g4.distance(self.p0)
 
+    @pytest.mark.skipif(
+        not (compat.USE_PYGEOS or compat.USE_SHAPELY_20),
+        reason="requires frechet_distance in shapely 2.0+",
+    )
+    def test_frechet_distance(self):
+        # closest point is (0, 0) in self.p1
+        expected = Series(
+            np.array([np.sqrt(5**2 + 5**2), np.nan]), self.na_none.index
+        )
+        assert_array_dtype_equal(expected, self.na_none.frechet_distance(self.p0))
+
+        expected = Series(
+            np.array([np.sqrt(5**2 + 5**2), np.nan]), self.na_none.index
+        )
+        assert_array_dtype_equal(expected, self.na_none.frechet_distance(self.p0))
+
+        expected = Series(np.array([np.nan, 0, 0, 0, 0, 0, np.nan, np.nan]), range(8))
+        with pytest.warns(UserWarning, match="The indices .+ different"):
+            assert_array_dtype_equal(
+                expected, self.g0.frechet_distance(self.g9, align=True)
+            )
+
+        val_1 = shapely.frechet_distance(self.g0.iloc[0], self.g9.iloc[0])
+        val_2 = shapely.frechet_distance(self.g0.iloc[2], self.g9.iloc[2])
+        val_3 = shapely.frechet_distance(self.g0.iloc[3], self.g9.iloc[3])
+        val_4 = shapely.frechet_distance(self.g0.iloc[4], self.g9.iloc[4])
```

Comment:
I would maybe hard-code these values. In this way, we are testing an implementation by using it again in tests which is not ideal.

Reply:
hardcoded the values